Update to fix issue with user and HttpContext

Sdhalm opened the pull request:
Issue with the way HttpContexts were gotten. Fixes ability to get actual context in system. Does not fix thread safe issue.

asbjornu commented on src/app/SharpRaven/Data/SentryRequestFactory.cs:
```diff
@@ -1,271 +1,288 @@
+            }
+        }
+    }
 }
```
It's not easy doing a review when the entire file has been changed.

asbjornu commented:
@Sdhalm Thanks for this! But can you please try to fix the broken tests and not modify the entire file, so reviewing becomes a bit easier? 😄

Sdhalm commented:
Just curious from an understanding point of view is there a reason why you allow for the setting of the HttpContext. From what I can see in the code package I believe this should only be a get.

Sdhalm commented on src/app/SharpRaven/Data/SentryRequestFactory.cs:
```diff
@@ -1,271 +1,288 @@
+    public class SentryRequestFactory : ISentryRequestFactory
+    {
+        [JsonIgnore]
+        internal static bool HasHttpContext
```
Change this over to perform a GetHttpContext in the event we don't have the context already.

Sdhalm commented on src/app/SharpRaven/Data/SentryRequestFactory.cs:
```diff
@@ -1,271 +1,288 @@
+        internal static dynamic HttpContext { get
+			{
+				try
+				{
```
Changed this to directly use the property we got from reflection instead of the auto property from before

Sdhalm commented on src/app/SharpRaven/Data/SentryRequestFactory.cs:
```diff
@@ -1,271 +1,288 @@
+
+        private static void TryGetHttpContextPropertyFromAppDomain()
+        {
+			if (HttpContext != null)
```
changed this to not use the property as this would cause an infinite loop/out of memory exception is doing so.

Sdhalm commented on src/app/SharpRaven/Data/SentryRequestFactory.cs:
```diff
@@ -1,271 +1,288 @@
+            {
+                TryGetHttpContextPropertyFromAppDomain();
+                CheckedForHttpContext = true;
+            }
```
Removed additional setting code and placed in needed respective locations in code, see other comments.

Sdhalm commented:
fixed the build issue related to the unit test. Although again this really shouldn't get a set at all based on the changes I had made. I also added comments to the files

asbjornu commented:
@Sdhalm Why did you close the issue?

Sdhalm commented:
I have a better update for it that I am working on. Including the unit tests.
On Jun 6, 2016 3:42 AM, Asbjørn Ulsberg <notifications@github.com> wrote:@Sdhalm Why did you close the issue?

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.

asbjornu commented:
@Sdhalm Ok, excellent. 👍  Looking forward to the pull request! 😄
